vcd/resource_vcd_nsxt_nat_rule.go

```diff
@@ -208,8 +240,24 @@ func resourceVcdNsxtNatRuleRead(ctx context.Context, d *schema.ResourceData, met
 
 func resourceVcdNsxtNatRuleDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	vcdClient := meta.(*VCDClient)
-	vcdClient.lockParentEdgeGtw(d)
-	defer vcdClient.unLockParentEdgeGtw(d)
+
+	// Handling locks is conditional. There are two scenarios:
```

Comment:
Probably too much to change it now, but shouldn't we extract this into a reusable function? As it's widely repeated. Maybe we can note it for the next release, what do you think?

Reply:
Yes. I also thought of the same, but just didn't want to introduce any possible errors at such a late stage. This locking is copy/pasted from other already verified and working functions

Reply:
Great, let’s do it for the next one then! 👍🏻